simtools/simtel/simtel_runner_array.py

```diff
@@ -121,92 +121,102 @@ def _load_simtel_data_directories(self):
+                    Array name.
+                label: str
+                    Instance label (optional).
+                mode: str
+                    out or err (optional, relevant only for sub_log).
+
+        Returns
+        -------
+        str
+            File name with full path.
 
         Raises
         ------
         ValueError
-            If run_number is not valid (not an unsigned int).
-
-        Returns
-        -------
-        Path:
-            Full path of the run log file.
+            If file_type is unknown.
         """
 
-        file_name = names.simtel_sub_log_file_name(
-            run=run_number,
-            primary=self.config.primary,
-            array_name=self.array_model.layout_name,
-            site=self.array_model.site,
-            zenith=self.config.zenith_angle,
-            azimuth=self.config.azimuth_angle,
-            label=self.label,
-            mode=mode,
+        file_label = (
+            f"_{kwargs['label']}" if "label" in kwargs and kwargs["label"] is not None else ""
         )
-        return self._simtel_log_dir.joinpath(file_name)
-
-    def get_histogram_file(self, run_number):
-        """Get full path of the simtel histogram file for a given run."""
-        file_name = names.simtel_histogram_file_name(
-            run=run_number,
-            primary=self.config.primary,
-            array_name=self.array_model.layout_name,
-            site=self.array_model.site,
-            zenith=self.config.zenith_angle,
-            azimuth=self.config.azimuth_angle,
-            label=self.label,
+        file_name = (
+            f"run{kwargs['run']}_{kwargs['primary']}_"
+            f"za{int(kwargs['zenith']):d}deg_azm{int(kwargs['azimuth']):d}deg_"
+            f"{kwargs['site']}_{kwargs['array_name']}{file_label}"
         )
-        return self._simtel_data_dir.joinpath(file_name)
-
-    def get_output_file(self, run_number):
-        """Get full path of the simtel output file for a given run."""
-        file_name = names.simtel_output_file_name(
-            run=run_number,
-            primary=self.config.primary,
-            array_name=self.array_model.layout_name,
-            site=self.array_model.site,
-            zenith=self.config.zenith_angle,
-            azimuth=self.config.azimuth_angle,
-            label=self.label,
-        )
-        return self._simtel_data_dir.joinpath(file_name)
+        if file_type == "log":
+            return self._simtel_log_dir.joinpath(f"{file_name}.log")
+        elif file_type == "histogram":
+            return self._simtel_log_dir.joinpath(f"{file_name}.hdata.zst")
+        elif file_type == "output":
+            return self._simtel_data_dir.joinpath(f"{file_name}.simtel.zst")
+        elif file_type == "sub_log":
+            suffix = ".log"
+            if "mode" in kwargs:
+                suffix = f".{kwargs['mode']}"
+            return self._simtel_log_dir.joinpath(f"log_sub_{file_name}{suffix}")
+        else:
```

Comment:
This else is not needed.

Reply:
Why is it not needed? Don't we want to raise an error if an unknown option is given?

Reply:
Because all other if statements have a return value (oh no - see now that `file_type` == sub_log does not always result in a return. I think that it is fine. 

But what should happen new mode is not in kwargs?

Reply:
All if statements have a return value, also the case of `file_type == "sub_log"`. The else is in case the `file_type` provided is not known. In that case I could either return a default file name (which I wouldn't know what it would be) or raise an error. I chose the latter.

> But what should happen new mode is not in kwargs?

if `mode` is not in kwargs then the suffix of the file is `.log` by default.

Reply:
OK - so you don't need the last else, as there is no decisions to be made. Anything, super small thing and some code linters will complain about it. Fine with me to keep it.

Reply:
Finally understood what you mean, removed it. Will wait for tests to pass and then merge.